src/dbnode/storage/index/read_through_segment.go

```diff
@@ -148,16 +148,18 @@ func (s *readThroughSegmentReader) MatchRegexp(
 		return s.Reader.MatchRegexp(field, c)
 	}
 
-	// TODO(rartoul): Would be nice not to allocate a string here.
-	pattern := c.FSTSyntax.String()
-	pl, ok := s.postingsListCache.GetRegexp(s.uuid, pattern)
+	plCacheQuery := PostingsListCacheQuery{
```

Comment:
Worth keeping the comment or is it not really viable to not allocate the string here any more?